### PR TITLE
Chain local secrets and defaults

### PR DESCRIPTION
**Pull request by patsplat**
Added support for configuring the DOTENV_FILE, and chaining to load both local secrets and defaults:

```bash
dotenv DOTENV_FILE=defaults.env dotenv run_my_command param1 param2 etc
```

### CODE REVIEW

**Comment by patsplat** [comment]
Noticed some other failures, going to change this to support a DOTENV_DEFAULT in one pass rather than chaining 

DOTENV_DEFAULT=defaults.env dotenv run_my_command param1 param2

Will also clean up the CI error

File: dotenv.sh [hunk 1]
```diff
@@ -30,30 +41,38 @@ export_envs() {
...
-	    fi
-	done < .env
+		fi
+	done < $1
 }
 
 # inject .env configs into the shell
-if [ -f .env ]; then
-	export_envs
+if [ -f ".env" ]; then
```
Comment (haolian9) [inline dotenv.sh:59]: Do you mean "$DOTENV_FILE"? Since the following `else` output a string contains `$DOTENV_FILE` ?

File: dotenv.sh [hunk 2]
```diff
@@ -30,30 +41,38 @@ export_envs() {
...
+	export_envs ".env"
 else
-	echo '.env file not found' >&2
+	echo '$DOTENV_FILE file not found'
```
Comment (haolian9) [inline dotenv.sh:62]: should it be double quote?

File: dotenv.sh [hunk 3]
```diff
@@ -1,12 +1,25 @@
...
 set -e
 
+log_verbose() {
+	if [ "$VERBOSE" = 1 ]; then
+		echo "[dotenv.sh] $1" >&2
+	fi
+}
+
+is_set() {
+	eval val=\""\$$1"\"
```
Comment (haolian9) [inline dotenv.sh:11]: could be `[ -n "$(eval "echo \$$1")" ]` concisely.

**Review by haolian9 — COMMENTED** [review]
Hi, @patsplat, when I merging from upstream to my fork, I found some issues.

File: dotenv.sh (hunk 3 above)
Comment (patsplat) [reply dotenv.sh:11]: Combining line 11 and 12 sounds good

File: dotenv.sh (hunk 1 above)
Comment (patsplat) [reply dotenv.sh:59]: yes :-(

File: dotenv.sh (hunk 2 above)
Comment (patsplat) [reply dotenv.sh:62]: yes :-(

**Comment by patsplat** [comment]
@haoliang-gao Implemented fixes in https://github.com/madcoda/dotenv-shell/pull/5

**Pull request merged**